src/OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule/ActivityHelper.cs

```diff
@@ -60,7 +63,25 @@ public static bool HasStarted(HttpContext context, out Activity? aspNetActivity)
     {
         PropagationContext propagationContext = textMapPropagator.Extract(default, context.Request, HttpRequestHeaderValuesGetter);
 
-        Activity? activity = AspNetSource.StartActivity(TelemetryHttpModule.AspNetActivityName, ActivityKind.Server, propagationContext.ActivityContext);
+        KeyValuePair<string, object?>[]? tags = null;
+        if (context.Request?.Unvalidated?.Path is string path)
+        {
+            tags = cachedTagsStorage;
+            if (tags is null)
+            {
+                tags = new KeyValuePair<string, object?>[1];
+                cachedTagsStorage = tags;
+            }
+
+            tags[0] = new KeyValuePair<string, object?>("url.path", path);
+        }
+
+        Activity? activity = AspNetSource.StartActivity(TelemetryHttpModule.AspNetActivityName, ActivityKind.Server, propagationContext.ActivityContext, tags);
+
+        if (tags is not null)
```

Comment:
is there a need to do this reset?

Reply:
Need? You could argue that it isn't essential to the functionality. 
I think it is a good idea to cleanup the value so that we don't have a reference to some random string living after it's been used. The code required to check and clear the field is a handful of assembly instructions.